### PR TITLE
Bug fix/40328 Timeout default value

### PR DESCRIPTION
**Pull request by GokulBothe**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [x] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [x] Sanity Tests are successfully executed for New and Existing Functionality
- [x] Verify that changes are compatible with all relevant browsers and platforms.
- [x] After creating pull request there should not be any conflicts
- [x] Resolve all Codacy comments
- [x] Builds and checks are passed before PR is sent for review
- [x] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
  - Enhanced timeout information tooltip for improved user guidance.
  
- **Bug Fixes**
  - Improved input validation on the timeout text box to prevent incorrect entries.
  - Timeout text box will now reset to `null` instead of `"0"` for clearer state management.

- **Chores**
  - Updated version numbers to reflect the latest release.

### CODE REVIEW

File: Ginger/Ginger/Actions/ActionEditPage.xaml.cs [hunk 1]
```diff
@@ -1668,7 +1668,7 @@ private void txtTimeout_TextChanged(object sender, TextChangedEventArgs e)
         {
             if (xTimeoutTextBox.Text == String.Empty || xTimeoutTextBox.Text == null)
             {
-                xTimeoutTextBox.Text = "0";
+                xTimeoutTextBox.Text = null;
```
Comment (Maheshkale447) [inline Ginger/Ginger/Actions/ActionEditPage.xaml.cs:1671]: Set it to empty instead of null

File: Ginger/Ginger/Actions/ActionEditPage.xaml.cs [hunk 2]
```diff
@@ -2259,5 +2259,34 @@ protected override void IsVisibleChangedHandler(object sender, DependencyPropert
...
+
+            bool isEmptyString = string.IsNullOrEmpty(e.Text);
+            if (isEmptyString)
+            {
+                //allow
+                e.Handled = false;
+                return;
+            }
+
+            string currentText = xTimeoutTextBox.Text != null ? xTimeoutTextBox.Text : string.Empty;
```
Comment (Maheshkale447) [inline Ginger/Ginger/Actions/ActionEditPage.xaml.cs:2274]: allow setting timeout to empty

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]
Please also check suggestion by CodeRabbit

File: Ginger/Ginger/Actions/ActionEditPage.xaml.cs (hunk 2 above)
Comment (GokulBothe) [reply Ginger/Ginger/Actions/ActionEditPage.xaml.cs:2274]: Yes it's allowing-
Basically this code is checking-
Allows input if the input text is empty.
Allows input if the resulting text (current text + new input) forms a valid integer.
Disallows any input that does not result in a valid integer.

**Pull request merged**